### DISCUSSION TITLE
how to make custom strategy for oauth 2.0?

### DISCUSSION THREAD

**Discussion by laptopmutia** [omniauth/omniauth#1081] [Q&A]
is this wiki still relevant https://github.com/omniauth/omniauth/wiki/Strategy-Contribution-Guide?

why its different if I compare it with [omniauth-facebook](https://github.com/simi/omniauth-facebook), [omniauth-twitter](https://github.com/arunagw/omniauth-twitter) and [omniauth-github](https://github.com/omniauth/omniauth-github/blob/master/lib/omniauth/strategies/github.rb) source code

nb: I found this https://github.com/omniauth/omniauth-oauth2

**Comment by pboling** [comment]
https://github.com/omniauth/omniauth-oauth2 is the answer!
